### PR TITLE
Onewire fixes

### PR DESCRIPTION
**Pull request by J-N-K**
this PR includes some minor fixes for multi sensors and a little refactoring of the multisensor discovery

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/device/DS2438.java [hunk 1]
```diff
@@ -43,6 +47,11 @@
  */
 @NonNullByDefault
 public class DS2438 extends AbstractOwDevice {
+    private final Logger logger = LoggerFactory.getLogger(DS2438.class);
+
+    private static final Pattern ASSOC_SENSOR_ID_PATTERN = Pattern
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/device/DS2438.java]: This seems to be unused.

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/discovery/OwDiscoveryItem.java [hunk 2]
```diff
@@ -151,16 +126,16 @@ public String getVendor() {
      *
      * @return
      */
```
Comment (htreu) [inline extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/discovery/OwDiscoveryItem.java]: Please complete the JavaDoc (@return annotation), also review the other methods.

**Comment by J-N-K** [comment]
Thanks, @htreu. I never saw that test failing locally.

File: extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/discovery/OwDiscoveryItem.java (hunk 2 above)
Comment (J-N-K) [reply extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/discovery/OwDiscoveryItem.java]: Yes. But I fixed the doc in another file where it was missing for the same method.

Comment (htreu) [reply extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/discovery/OwDiscoveryItem.java]: Are you going to complete it here, too? it looks half finished like this.

Comment (J-N-K) [reply extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/discovery/OwDiscoveryItem.java]: Sure.

**Pull request merged**